pkg/pgclient/client.go

```diff
@@ -66,15 +66,26 @@ type Client struct {
 	metricCache   *pgmodel.MetricNameCache
 }
 
+// Post connect validation function, useful for things such as acquiring locks
+// that should live the duration of the connection
+type LockFunc = func(ctx context.Context, conn *pgx.Conn) error
+
 // NewClient creates a new PostgreSQL client
-func NewClient(cfg *Config) (*Client, error) {
+func NewClient(cfg *Config, schemaLocker LockFunc) (*Client, error) {
 	connectionStr := cfg.GetConnectionStr()
 	minConnections, maxConnections, numCopiers, err := cfg.GetNumConnections()
 	if err != nil {
-		log.Error("err configuring number of connections", util.MaskPassword(err.Error()))
+		log.Error("msg", "configuring number of connections", "err", util.MaskPassword(err.Error()))
```

Comment:
dunno, and it can change over time, so better safe (anything that transitively uses the CFG can potentially report the password needs to be assumed to report sensitive info, which gets to be nearly everything in client)